mmocr/models/ner/utils/bert.py

```diff
@@ -0,0 +1,488 @@
+# ------------------------------------------------------------------------------
+# Adapted from https://github.com/lonePatient/BERT-NER-Pytorch
+# Original licence: Copyright (c) 2020 Weitang Liu, under the MIT License.
+# ------------------------------------------------------------------------------
+
+import math
+
+import torch
+import torch.nn as nn
+
+from mmocr.models.ner.utils.activations import ACT2FN
+
+
+class BertModel(nn.Module):
+    """Implement Bert model for named entity recognition task.
+
+    The code is adapted from https://github.com/lonePatient/BERT-NER-Pytorch
+    Args:
+        num_hidden_layers (int): The number of hidden layers.
+        initializer_range (float):
+        vocab_size (int): Number of words supported.
+        hidden_size (int): Hidden size.
+        max_position_embeddings (int): Max positionsembedding size.
+        type_vocab_size (int): The size of type_vocab.
+        layer_norm_eps (float): eps.
+        hidden_dropout_prob (float): The dropout probability of hidden layer.
+        output_attentions (bool):  Whether use the attentions in output
+        output_hidden_states (bool): Whether use the hidden_states in output.
+        num_attention_heads (int): The number of attention heads.
+        attention_probs_dropout_prob (float): The dropout probability
+            of attention.
+        intermediate_size (int): The size of intermediate layer.
+        hidden_act (str):  hidden layer activation
+    """
+
+    def __init__(self,
+                 num_hidden_layers=12,
+                 initializer_range=0.02,
+                 vocab_size=21128,
+                 hidden_size=768,
+                 max_position_embeddings=128,
+                 type_vocab_size=2,
+                 layer_norm_eps=1e-12,
+                 hidden_dropout_prob=0.1,
+                 output_attentions=False,
+                 output_hidden_states=False,
+                 num_attention_heads=12,
+                 attention_probs_dropout_prob=0.1,
```

Comment:
I don't understand the name `attention_probs_dropout_prob`. Why there are two probs?